src/api/getLanglinks.js

```diff
@@ -16,34 +21,37 @@ export const getLanglinks = (lang, title) => {
   return cachedFetch(url, response => {
     const { pages } = response.query
     const langlinks = pages[0].langlinks
-    const prioritizedLanguages = langlinks
-      .filter(item => {
-        return prioritizedList.indexOf(item.lang) !== -1 && isSupportedForReading(item.lang)
-      })
+    const allLanguages = langlinks
       .map(item => {
         return {
           title: item.autonym,
           langname: item.langname,
-          lang: item.lang,
+          lang: convertToWikiLang(item.lang),
           description: item.title,
           dir: getDirection(item.lang)
         }
       })
-    const allLanguages = prioritizedLanguages.concat(
-      langlinks
-        .filter(item => {
-          return isSupportedForReading(item.lang) && prioritizedList.indexOf(item.lang) === -1
-        })
-        .map(item => {
-          return {
-            title: item.autonym,
-            langname: item.langname,
-            lang: item.lang,
-            description: item.title,
-            dir: getDirection(item.lang)
-          }
-        })
-    )
+      .filter(item => isSupportedForReading(item.lang))
+
+    // Sort languages prioritized first, and then
+    // in alphabetical order within the prioritized
+    // and non-prioritized groups.
+    allLanguages.sort((a, b) => {
```

Comment:
in app language setting, we have the different sort algorithm, should we do the same there?

Reply:
The main difference is over there, both `prioritizedLanguages` and `allLanguages` are used so it makes sense to build them separately.